### PR TITLE
Core Data: Batch remaining actions in resolvers

### PR DESCRIPTION
**Pull request by Mamaduka**
## What?
Similar to #60591.
Supersedes and closes #60604.

PR batches multiple actions dispatched by `getNavigationFallbackId` and `getRevisions` resolvers.

### Props
```
Co-authored-by: ellatrix <ellatrix@git.wordpress.org>
```

## Testing Instructions
* CI checks should be green.

### CODE REVIEW

File: packages/core-data/src/resolvers.js [hunk 1]
```diff
@@ -884,32 +888,36 @@ export const getRevisions =
...
+					const resolutionsArgs = records
+						.filter( ( record ) => record[ key ] )
+						.map( ( record ) => [
+							kind,
+							name,
+							recordKey,
+							record[ key ],
+						] );
 
-				dispatch.startResolutions( 'getRevision', resolutionsArgs );
```
Comment (getdave) [inline packages/core-data/src/resolvers.js:910]: How is it that we can now remove this?

File: packages/core-data/src/resolvers.js [hunk 2]
```diff
@@ -764,9 +764,13 @@ export const getNavigationFallbackId =
 
 		const record = fallback?._embedded?.self;
 
-		dispatch.receiveNavigationFallbackId( fallback?.id );
+		registry.batch( () => {
+			dispatch.receiveNavigationFallbackId( fallback?.id );
+
+			if ( ! record ) {
```
Comment (getdave) [inline packages/core-data/src/resolvers.js:770]: Prefer exit early is 👍

**Review by getdave — APPROVED** [review]
Thanks for improving this.

File: packages/core-data/src/resolvers.js (hunk 1 above)
Comment (Mamaduka) [reply packages/core-data/src/resolvers.js:910]: See: https://github.com/WordPress/gutenberg/pull/63430#discussion_r1673992148

Comment (getdave) [reply packages/core-data/src/resolvers.js:910]: TIL - thanks.

**Comment by Mamaduka** [comment]
Thanks for testing, @getdave!

**Pull request merged**